rust/windows-client/src-tauri/src/client/gui.rs

```diff
@@ -395,15 +403,18 @@ impl Controller {
 async fn run_controller(
     app: tauri::AppHandle,
     mut rx: mpsc::Receiver<ControllerRequest>,
+    logging_handles: client::logging::Handles,
+    advanced_settings: AdvancedSettings,
 ) -> Result<()> {
-    let mut controller = match Controller::new(app.clone()).await {
-        Err(e) => {
-            // TODO: There must be a shorter way to write these?
-            tracing::error!("couldn't create controller: {e}");
-            return Err(e);
-        }
-        Ok(x) => x,
-    };
+    let mut controller =
+        match Controller::new(app.clone(), logging_handles, advanced_settings).await {
```

Comment:
can we just use `?` here and log on the top level function?

Reply:
Yeah, it might be better if I do `.context` or something with anyhow. So there will be a descriptive error "failed to create controller" but without a whole match statement.